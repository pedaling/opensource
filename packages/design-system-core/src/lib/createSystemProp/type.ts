import type { Theme } from '../types';

export type InterpolationFn = (props: Record<string, any>) => Record<string, any>;

export type SystemPropConfig = {
  property: string;
  styleProperty?: string;
  scale?: keyof Pick<Theme, 'colors' | 'space' | 'opacity'>;
  disabled?: boolean;
  transform?: (value: any) => Record<string, any> | undefined;
};

export type SystemProp = {
  (props: Record<string, any>): Record<string, any>;
  propName: string;
  disabled: boolean;
};
