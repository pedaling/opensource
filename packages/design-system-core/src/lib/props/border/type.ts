import type { ResponsiveValue } from '../../types';

export type BorderProps = {
  borderWidth?: ResponsiveValue<number>;
  borderStyle?: ResponsiveValue<'solid' | 'none'>;
  borderColor?: ResponsiveValue<string>;
  borderTopWidth?: ResponsiveValue<number>;
  borderTopStyle?: ResponsiveValue<'solid' | 'none'>;
  borderTopColor?: ResponsiveValue<string>;
  borderRightWidth?: ResponsiveValue<number>;
  borderRightStyle?: ResponsiveValue<'solid' | 'none'>;
  borderRightColor?: ResponsiveValue<string>;
  borderBottomWidth?: ResponsiveValue<number>;
  borderBottomStyle?: ResponsiveValue<'solid' | 'none'>;
  borderBottomColor?: ResponsiveValue<string>;
  borderLeftWidth?: ResponsiveValue<number>;
  borderLeftStyle?: ResponsiveValue<'solid' | 'none'>;
  borderLeftColor?: ResponsiveValue<string>;
  borderRadius?: ResponsiveValue<number | string>;
  borderTopLeftRadius?: ResponsiveValue<number | string>;
  borderTopRightRadius?: ResponsiveValue<number | string>;
  borderBottomLeftRadius?: ResponsiveValue<number | string>;
  borderBottomRightRadius?: ResponsiveValue<number | string>;
  outlineWidth?: ResponsiveValue<number>;
  outlineStyle?: ResponsiveValue<'solid' | 'none'>;
  outlineColor?: ResponsiveValue<string>;
  outlineOffset?: ResponsiveValue<number>;
};
