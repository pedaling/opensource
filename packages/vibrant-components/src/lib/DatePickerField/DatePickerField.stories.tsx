import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { Body } from '../Body';
import { Pressable } from '../Pressable';
import { Space } from '../Space';
import { VStack } from '../VStack';
import { DatePickerField } from './DatePickerField';

export default {
  title: 'DatePickerField',
  component: DatePickerField,
  args: {
    label: '날짜 선택',
  },
} as ComponentMeta<typeof DatePickerField>;

export const Basic: ComponentStory<typeof DatePickerField> = props => (
  <VStack width="100%">
    <Space height={450} />
    <DatePickerField {...props} />
    <Space height={450} />
  </VStack>
);

export const WithForm: ComponentStory<typeof DatePickerField> = props => (
  <form
    style={{ width: '100%' }}
    onSubmit={event => {
      console.log('submit');

      event.preventDefault();
    }}
  >
    <Space height={450} />
    <DatePickerField {...props} />
    <Space height={50} />
    <Pressable
      type="submit"
      width="100%"
      backgroundColor="primary"
      py={15}
      borderRadiusLevel={1}
      overlayColor="onPrimary"
      interactions={['hover', 'focus', 'active']}
    >
      <Body color="onPrimary" textAlign="center" level={1} weight="bold">
        submit
      </Body>
    </Pressable>
    <Space height={450} />
  </form>
);
