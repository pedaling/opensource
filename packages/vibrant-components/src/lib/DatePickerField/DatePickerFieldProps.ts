import type { ForwardedRef } from 'react';
import { withVariation } from '@vibrant-ui/core';
import type { BaseInputProps } from '../../types';

export type DatePickerFieldRefValue = {
  focus: (_: { preventScroll?: boolean }) => void;
  blur: () => void;
};

export type DatePickerFieldProps = BaseInputProps<Date | undefined> & {
  ref?: ForwardedRef<DatePickerFieldRefValue>;
  helperText?: string;
  state?: 'default' | 'error';
  onOpen?: () => void;
  zIndex?: number;
  autoFocus?: boolean;
  testId?: string;
} & (
    | {
        label?: never;
        placeholder?: string;
      }
    | {
        label?: string;
        placeholder?: never;
      }
  );

export const withDatePickerFieldVariation = withVariation<DatePickerFieldProps>('DatePickerField')();
