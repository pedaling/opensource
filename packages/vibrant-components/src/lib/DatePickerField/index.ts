export { DatePickerField } from './DatePickerField';
export type { DatePickerFieldProps, DatePickerFieldRefValue } from './DatePickerFieldProps';
