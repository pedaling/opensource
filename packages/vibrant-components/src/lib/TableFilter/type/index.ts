export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
