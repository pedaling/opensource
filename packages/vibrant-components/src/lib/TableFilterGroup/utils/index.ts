import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
