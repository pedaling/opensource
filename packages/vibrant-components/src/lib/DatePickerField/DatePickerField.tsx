import { useEffect, useState } from 'react';
import { Box } from '@vibrant-ui/core';
import { toZeroPadded, useSafeDeps } from '@vibrant-ui/utils';
import { Body } from '../Body';
import { DatePicker } from '../DatePicker';
import { Pressable } from '../Pressable';
import { withDatePickerFieldVariation } from './DatePickerFieldProps';

export const DatePickerField = withDatePickerFieldVariation(({ defaultValue, onValueChange, id, ...restProps }) => {
  const [value, setValue] = useState(defaultValue);
  const [isPickerOpened, setIsPickerOpened] = useState(false);
  const [inputValue, setInputValue] = useState(defaultValue?.toDateString() ?? '');
  const onValueChangeRef = useSafeDeps(onValueChange);

  const getDateString = (date: Date) =>
    `${date.getFullYear()}-${toZeroPadded(date.getMonth() + 1, 2)}-${toZeroPadded(date.getDate(), 2)}`;

  useEffect(() => {
    if (!defaultValue) {
      return;
    }

    setValue(defaultValue);
  }, [defaultValue]);

  useEffect(() => {
    if (!value) {
      return;
    }

    onValueChangeRef.current?.(value);

    setInputValue(getDateString(value));
  }, [onValueChangeRef, value]);

  return (
    <Box position="relative" width="100%" height={50}>
      <Box<typeof Pressable>
        base={Pressable}
        width="100%"
        height="100%"
        borderWidth={1}
        borderColor="outline1"
        typography="body2"
        p={16}
        backgroundColor="surface3"
        onFocus={() => setIsPickerOpened(true)}
        onBlur={() => setIsPickerOpened(false)}
        {...restProps}
      >
        <Body level={2}>{inputValue}</Body>
      </Box>
      {isPickerOpened && (
        <Box
          base={Pressable}
          position="absolute"
          onFocus={() => setIsPickerOpened(true)}
          onBlur={() => setIsPickerOpened(false)}
          top={56}
          left={0}
        >
          <DatePicker
            range={false}
            date={value}
            onDateSelect={selectedDate => {
              setValue(selectedDate);

              setIsPickerOpened(false);
            }}
          />
        </Box>
      )}
    </Box>
  );
});
