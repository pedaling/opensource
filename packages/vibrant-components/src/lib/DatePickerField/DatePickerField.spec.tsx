import type { ReactRenderer } from '@vibrant-ui/utils/testing-web';
import { createReactRenderer } from '@vibrant-ui/utils/testing-web';
import { DatePickerField } from './DatePickerField';

describe('<DatePickerField />', () => {
  const { render } = createReactRenderer();

  let renderer: ReactRenderer;

  describe('when size is lg', () => {
    beforeEach(async () => {
      renderer = render(<DatePickerField size="lg" label="label" />);
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });

  describe('when size is md', () => {
    beforeEach(async () => {
      renderer = render(<DatePickerField size="md" label="label" />);
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });

  describe('when size is sm', () => {
    beforeEach(async () => {
      renderer = render(<DatePickerField size="sm" label="label" />);
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });
});
