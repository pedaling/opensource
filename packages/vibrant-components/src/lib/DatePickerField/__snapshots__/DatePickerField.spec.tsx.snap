// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePickerField /> when size is lg match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  border-width: 1px;
  border-style: solid;
  border-color: #00000026;
  height: 50px;
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ffffff;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  background-color: inherit;
  border-width: 0;
  padding: 0;
  overflow: hidden;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  z-index: 0;
  height: 100%;
  padding-left: 15px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-4 {
  text-align: left;
  color: #5b5b5b;
  -webkit-text-fill-color: #5b5b5b;
  text-fill-color: #5b5b5b;
  display: -webkit-box;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-weight: 400;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  padding-left: 12px;
  height: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  background-color: inherit;
  border-width: 0;
  padding: 0;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding-right: 15px;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 20px;
  height: 20px;
  fill: #5b5b5b;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
}

<div>
  <div
    class="emotion-0"
    data-testid="date-picker-field"
  >
    <div
      class="emotion-0"
      data-testid="date-input"
    >
      <div
        class="emotion-2"
      >
        <button
          class="emotion-3"
          data-testid="pressable"
          style="opacity: 1; transition-property: opacity; transition-duration: 200ms; transition-timing-function: cubic-bezier(0.5, 1, 0.89, 1);"
          type="button"
        >
          <span
            class="emotion-4"
            data-testid="body"
          >
            label
          </span>
        </button>
        <div
          class="emotion-5"
        >
          <button
            aria-label="날짜 선택"
            class="emotion-6"
            type="button"
          >
            <svg
              class="emotion-7"
              data-testid="calendar-thin"
              viewBox="0 0 24 24"
            >
              <path
                class="emotion-8"
                d="M17.25 13.5h-3.5c-.15 0-.25.1-.25.25v3.5c0 .15.1.25.25.25h3.5c.15 0 .25-.1.25-.25v-3.5c0-.15-.1-.25-.25-.25Z"
              />
              <path
                class="emotion-8"
                d="M21.75 4h-4.6V2.25C17.1 2.1 17 2 16.9 2h-1.25c-.15 0-.25.1-.25.25V4H8.6V2.25c0-.15-.1-.25-.2-.25H7.15c-.15 0-.25.1-.25.25V4H2.25C2.1 4 2 4.1 2 4.25v17.5c0 .15.1.25.25.25h19.5c.15 0 .25-.1.25-.25V4.25c0-.15-.1-.25-.25-.25Zm-1.5 16.25H3.75V5.75h3.1V7.5c0 .15.1.25.25.25h1.25c.15 0 .25-.1.25-.25V5.75h6.75V7.5c0 .15.1.25.25.25h1.25c.15 0 .25-.1.25-.25V5.75h3.1v14.5h.05Z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<DatePickerField /> when size is md match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  border-width: 1px;
  border-style: solid;
  border-color: #00000026;
  height: 38px;
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ffffff;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  background-color: inherit;
  border-width: 0;
  padding: 0;
  overflow: hidden;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  z-index: 0;
  height: 100%;
  padding-left: 9px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-4 {
  text-align: left;
  color: #5b5b5b;
  -webkit-text-fill-color: #5b5b5b;
  text-fill-color: #5b5b5b;
  display: -webkit-box;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-weight: 400;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  padding-left: 8px;
  height: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  background-color: inherit;
  border-width: 0;
  padding: 0;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding-right: 9px;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 20px;
  height: 20px;
  fill: #5b5b5b;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
}

<div>
  <div
    class="emotion-0"
    data-testid="date-picker-field"
  >
    <div
      class="emotion-0"
      data-testid="date-input"
    >
      <div
        class="emotion-2"
      >
        <button
          class="emotion-3"
          data-testid="pressable"
          style="opacity: 1; transition-property: opacity; transition-duration: 200ms; transition-timing-function: cubic-bezier(0.5, 1, 0.89, 1);"
          type="button"
        >
          <span
            class="emotion-4"
            data-testid="body"
          >
            label
          </span>
        </button>
        <div
          class="emotion-5"
        >
          <button
            aria-label="날짜 선택"
            class="emotion-6"
            type="button"
          >
            <svg
              class="emotion-7"
              data-testid="calendar-thin"
              viewBox="0 0 24 24"
            >
              <path
                class="emotion-8"
                d="M17.25 13.5h-3.5c-.15 0-.25.1-.25.25v3.5c0 .15.1.25.25.25h3.5c.15 0 .25-.1.25-.25v-3.5c0-.15-.1-.25-.25-.25Z"
              />
              <path
                class="emotion-8"
                d="M21.75 4h-4.6V2.25C17.1 2.1 17 2 16.9 2h-1.25c-.15 0-.25.1-.25.25V4H8.6V2.25c0-.15-.1-.25-.2-.25H7.15c-.15 0-.25.1-.25.25V4H2.25C2.1 4 2 4.1 2 4.25v17.5c0 .15.1.25.25.25h19.5c.15 0 .25-.1.25-.25V4.25c0-.15-.1-.25-.25-.25Zm-1.5 16.25H3.75V5.75h3.1V7.5c0 .15.1.25.25.25h1.25c.15 0 .25-.1.25-.25V5.75h6.75V7.5c0 .15.1.25.25.25h1.25c.15 0 .25-.1.25-.25V5.75h3.1v14.5h.05Z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<DatePickerField /> when size is sm match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  border-width: 1px;
  border-style: solid;
  border-color: #00000026;
  height: 30px;
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ffffff;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  background-color: inherit;
  border-width: 0;
  padding: 0;
  overflow: hidden;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  z-index: 0;
  height: 100%;
  padding-left: 7px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-4 {
  text-align: left;
  color: #5b5b5b;
  -webkit-text-fill-color: #5b5b5b;
  text-fill-color: #5b5b5b;
  display: -webkit-box;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
  font-size: 0.75rem;
  line-height: 1rem;
  font-weight: 400;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  padding-left: 6px;
  height: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  background-color: inherit;
  border-width: 0;
  padding: 0;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding-right: 7px;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 16px;
  height: 16px;
  fill: #5b5b5b;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
}

<div>
  <div
    class="emotion-0"
    data-testid="date-picker-field"
  >
    <div
      class="emotion-0"
      data-testid="date-input"
    >
      <div
        class="emotion-2"
      >
        <button
          class="emotion-3"
          data-testid="pressable"
          style="opacity: 1; transition-property: opacity; transition-duration: 200ms; transition-timing-function: cubic-bezier(0.5, 1, 0.89, 1);"
          type="button"
        >
          <span
            class="emotion-4"
            data-testid="body"
          >
            label
          </span>
        </button>
        <div
          class="emotion-5"
        >
          <button
            aria-label="날짜 선택"
            class="emotion-6"
            type="button"
          >
            <svg
              class="emotion-7"
              data-testid="calendar-thin"
              viewBox="0 0 24 24"
            >
              <path
                class="emotion-8"
                d="M17.25 13.5h-3.5c-.15 0-.25.1-.25.25v3.5c0 .15.1.25.25.25h3.5c.15 0 .25-.1.25-.25v-3.5c0-.15-.1-.25-.25-.25Z"
              />
              <path
                class="emotion-8"
                d="M21.75 4h-4.6V2.25C17.1 2.1 17 2 16.9 2h-1.25c-.15 0-.25.1-.25.25V4H8.6V2.25c0-.15-.1-.25-.2-.25H7.15c-.15 0-.25.1-.25.25V4H2.25C2.1 4 2 4.1 2 4.25v17.5c0 .15.1.25.25.25h19.5c.15 0 .25-.1.25-.25V4.25c0-.15-.1-.25-.25-.25Zm-1.5 16.25H3.75V5.75h3.1V7.5c0 .15.1.25.25.25h1.25c.15 0 .25-.1.25-.25V5.75h6.75V7.5c0 .15.1.25.25.25h1.25c.15 0 .25-.1.25-.25V5.75h3.1v14.5h.05Z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
